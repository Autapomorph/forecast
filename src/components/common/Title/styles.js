import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const StyledTitle = styled.h3`
  color: var(--base-text-color);
  font-size: 1.3rem;
  font-weight: bold;
`;
