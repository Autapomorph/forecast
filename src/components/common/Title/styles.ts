import styled from 'styled-components/macro';

export const Title = styled.span`
  margin: 0;
  font-size: 1.3rem;
  font-weight: bold;
`;

export const Subtitle = styled.span`
  font-size: 1rem;
  font-weight: normal;
`;
