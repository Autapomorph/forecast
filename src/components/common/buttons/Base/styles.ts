import styled from 'styled-components/macro';

export const IconButton = styled.button.attrs({
  type: 'button',
})`
  padding: 5px;
  border: none;
  background: none;
  color: inherit;
  cursor: pointer;

  :hover {
    color: var(--white-dark);
  }
`;
