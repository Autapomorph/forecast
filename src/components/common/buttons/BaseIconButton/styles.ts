import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const StyledIconButton = styled.span`
  cursor: pointer;

  :hover {
    color: var(--light-gray);
  }
`;
