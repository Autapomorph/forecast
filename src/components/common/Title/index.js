import React from 'react';

import { StyledTitle, StyledSubtitle } from './styles';

const Title = ({ children }) => <StyledTitle>{children}</StyledTitle>;

Title.Subtitle = ({ children }) => <StyledSubtitle>{children}</StyledSubtitle>;

export default Title;
