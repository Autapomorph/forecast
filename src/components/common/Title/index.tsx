import React from 'react';

import { StyledTitle, StyledSubtitle } from './styles';

interface IProps extends React.FC {
  Subtitle: React.FC;
}

const Title: IProps = ({ children }): React.ReactElement => <StyledTitle>{children}</StyledTitle>;

Title.Subtitle = ({ children }): React.ReactElement => <StyledSubtitle>{children}</StyledSubtitle>;

export default Title;
