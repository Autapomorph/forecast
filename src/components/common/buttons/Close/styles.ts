import styled from 'styled-components/macro';

import { ThemeProp } from 'models';
import BaseIconButton from 'components/common/buttons/Base';

export const CloseButton = styled(BaseIconButton)`
  padding: 5px 10px;
  color: ${({ theme }: ThemeProp) => theme.baseTextColor};
  font-size: 1.2rem;

  :hover {
    color: var(--gray-light-lighter);
  }
`;
