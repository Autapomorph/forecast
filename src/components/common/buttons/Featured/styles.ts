import styled from 'styled-components/macro';

import BaseIconButton from 'components/common/buttons/Base';

export const FeaturedButton = styled(BaseIconButton)`
  color: var(--yellow);

  :hover {
    color: var(--yellow-dark);
  }
`;
