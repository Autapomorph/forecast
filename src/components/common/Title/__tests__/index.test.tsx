import React from 'react';
import { shallow } from 'enzyme';

import Title from '..';

import * as S from '../styles';

describe('Title tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<Title>Test</Title>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render children', () => {
    const text = 'text';
    const wrapper = shallow(<Title>{text}</Title>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(S.Title).text()).toEqual(text);
  });

  it('should render SubTitle', () => {
    const text = 'text';
    const wrapper = shallow(<Title.Subtitle>{text}</Title.Subtitle>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(S.Subtitle).text()).toEqual(text);
  });
});
