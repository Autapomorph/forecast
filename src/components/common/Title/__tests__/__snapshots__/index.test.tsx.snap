// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title tests renders properly 1`] = `
<styles__Title>
  Test
</styles__Title>
`;

exports[`Title tests should render SubTitle 1`] = `
<styles__Subtitle>
  text
</styles__Subtitle>
`;

exports[`Title tests should render children 1`] = `
<styles__Title>
  text
</styles__Title>
`;
